### PR TITLE
Prepare poms for release

### PR DESCRIPTION
**Pull request by miszobi**
This:
* sets up maven-release-plugin to release to central through [Sonatype's OSS repository hosting](http://central.sonatype.org/pages/ossrh-guide.html)
* adds metadata to the poms [required](http://central.sonatype.org/pages/requirements.html) for releasing to central
* configures [signing the released artifacts](http://central.sonatype.org/pages/apache-maven.html#gpg-signed-components) using gpg and maven-gpg-plugin
* cleans up some duplicated properties/unneded dependencies

### One-time setup before releasing
* Set up an account with Sonatype as documented [here](http://central.sonatype.org/pages/ossrh-guide.html#create-a-ticket-with-sonatype)
* Make sure the ticket is marked as Resolved before proceeding
* Add the account credentials to your `$HOME/.m2/settings.xml`:
```
<settings>
...
   <servers>
     <server>
       <id>sonatype-nexus-staging</id>
       <username>sonatype-nexus-userid</username>
       <password>sonatype-nexus-password</password>
     </server>
     <server>
       <id> sonatype-nexus-snapshots</id>
       <username>sonatype-nexus-userid</username>
       <password>sonatype-nexus-password</password>
     </server>
...
```
* install gpg, generate a key-pair (`gpg --gen-key`, then `gpg2 --list-keys` to get your key id), send your public key to the keyserver (`gpg2 --keyserver hkp://pool.sks-keyservers.net --send-keys KEYID`) - see [detailed instructions](http://central.sonatype.org/pages/working-with-pgp-signatures.html)

### Doing the release

Run `mvn clean release:prepare release:perform`.
* This will ask for the version number to be released, and the next development version number, defaults should be fine
* it'll ask a number of times for your gpg key passphrase to sign all the artifacts
* tag the release in git and increment the version number
* deploy the artifacts to sonatype

Artifacts will be deployed to the sonatype staging area. Go to https://oss.sonatype.org/ and follow the [instructions](http://central.sonatype.org/pages/releasing-the-deployment.html) to locate and close the staging repository for the release, to release the artifacts to central.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -8,16 +7,9 @@
 	</parent>
 
 	<packaging>jar</packaging>
-	<name>Onelogin java-saml Toolkit Core</name>
+	<name>OneLogin java-saml Toolkit Core</name>
 	<artifactId>java-saml</artifactId>
 
-	<properties>
```
Comment (miszobi) [inline core/pom.xml]: Inherited from parent, were duplicated here.

File: core/pom.xml [hunk 2]
```diff
@@ -53,9 +45,15 @@
 
 		<!-- for log -->
 		<dependency>
```
Comment (miszobi) [inline core/pom.xml]: In general you want to only include the slf4j api, not the implementation. I've left the implementation for tests - http://slf4j.org/faq.html#maven2.

File: pom.xml [hunk 3]
```diff
@@ -60,7 +62,107 @@
...
+			<name>Sonatype Nexus Snapshots</name>
+			<url>https://oss.sonatype.org/content/repositories/snapshots/</url>
+		</snapshotRepository>
+	</distributionManagement>
+
+	<scm>
+		<connection>scm:git:git@github.com:onelogin/java-saml.git</connection>
+		<developerConnection>scm:git:git@github.com:onelogin/java-saml.git</developerConnection>
+		<url>https://github.com/onelogin/java-saml</url>
+		<tag>HEAD</tag>
```
Comment (pitbulk) [inline pom.xml]: Maybe reference for a v2.0.0 tag that I can create in the same way I generated the [v.1.1.2](https://github.com/onelogin/java-saml/releases/tag/v1.1.2)?

**Comment by pitbulk** [comment]
Thanks for this contribution. I will follow your steps.

**Comment by pitbulk** [comment]
I created the Jira account and the project: https://issues.sonatype.org/browse/OSSRH-24376

**Comment by pitbulk** [comment]
The ticket is now marked as "RESOLVED" but before release I want to provide to the toolkit some [documentation](https://github.com/onelogin/java-saml/issues/49)

File: pom.xml (hunk 3 above)
Comment (luuuis) [inline pom.xml]: The maven-release-plugin will create the tag for you when you run `mvn release:prepare` down the road (it also updates this file in the tagged commit).

**Pull request merged**